backend/api/views/quiz.py

```diff
@@ -1,80 +1,77 @@
-            insertion_result = self.__quizzes.insert_one(quiz_data)
-            inserted_quiz = self.__quizzes.find_one(
+            insertion_result = self._quizzes.insert_one(quiz_data)
+            inserted_quiz = self._quizzes.find_one(
                 {"_id": insertion_result.inserted_id}
             )
             return JsonResponse(data=inserted_quiz, status=status.HTTP_201_CREATED)
 
 
-class QuizDetails(APIView):
-    __quizzes = client.collection("quizzes")
-    __validator = SchemaValidator("quizzes")
+class QuizDetail(QuizApiView):
+    """GET, PATCH and DELETE a single quiz."""
 
-    def get(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "GET":
-            quiz = self.__quizzes.find_one({"id": pk})
+    def get(self, request: Request, pk: str):
+        quiz = self._quizzes.find_one({"id": pk})
 
-            if quiz is None:
-                raise Http404("There is no quiz matching your query.")
-            else:
-                return JsonResponse(data=quiz)
+        if quiz is None:
+            raise Http404("There is no quiz matching your query.")
+        else:
+            return JsonResponse(data=quiz)
 
     def patch(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "PATCH":
-            quiz_data = request.data
+        quiz_data = request.data
 
-            try:
-                self.__validator.validate(quiz_data)
-            except InvalidData as error:
-                return JsonResponse(
-                    data={"message": f"Invalid JSON: {str(error)}"},
-                    status=status.HTTP_400_BAD_REQUEST,
-                )
-            else:
-                updated_quiz = self.__quizzes.find_one_and_update(
-                    {"id": pk},
-                    {"$set": quiz_data},
-                    # If the document is not already in collection, insert it
-                    upsert=True,
-                    # Return the updated document
-                    return_document=ReturnDocument.AFTER,
-                )
-                return JsonResponse(data=updated_quiz)
+        try:
+            self._validator.validate(quiz_data)
+        except InvalidData as error:
+            return JsonResponse(
+                data={"message": f"Invalid JSON: {str(error)}"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        else:
+            updated_quiz = self._quizzes.find_one_and_update(
+                {"id": pk},
+                {"$set": quiz_data},
+                # Return the updated document
+                return_document=ReturnDocument.AFTER,
+            )
+            return JsonResponse(data=updated_quiz)
 
     def delete(self, request: Request, pk: str) -> JsonResponse:
-        if request.method == "DELETE":
-            self.__quizzes.find_one_and_delete({"id": pk})
-            return JsonResponse(
-                data={}
-            )  # An empty JSON is returned, as to confirm deletion
```

Comment:
It's best practice to return the `id` of the item deleted. But an empty dict is also fine